### PR TITLE
[Fix][WRS-1860] Prevent publishing of connector without explicitly specified allowedDomains options

### PR DESCRIPTION
**Pull request by psamusev**
In context of this PR we did in addition:
- Specify `-v` shortcut to log version of CLI tool
- Refactor `publish` according to `set-auth` command structure
- Improve `verbose` logging

### CODE REVIEW

File: src/connector-cli/src/commands/publish/steps/validate-allowed-domains.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  }
+
+  verbose('Validating domains: ' + JSON.stringify(domains));
+
+  const allDomains = domains.filter((d) => d === '*');
+  if (allDomains.length > 0) {
+    verbose('Incorrect domains: ' + JSON.stringify(allDomains));
+    throw new ExecutionError(
+      'Some of "allowedDomains" uses "*" as value. This is prohibited. You can specify "*" only as part of host. Execute command with --verbose for more information or use -h for command for details about arguments'
+    );
```
Comment (Matthiee) [inline src/connector-cli/src/commands/publish/steps/validate-allowed-domains.ts:18]:
```suggestion
      'Some "allowedDomains" use "*" as the entire value, which is prohibited. You can only use "*" as part of a host. Execute command with --verbose for more information or use -h for command for details about arguments'
    );
```

File: src/connector-cli/src/commands/publish/steps/validate-allowed-domains.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  }
+  const useProtocol = domains.filter(
+    (d) => d.startsWith('http://') || d.startsWith('https://')
+  );
+  if (useProtocol.length > 0) {
+    verbose('Incorrect domains: ' + JSON.stringify(useProtocol));
+    throw new ExecutionError(
+      'Some of "allowedDomains" starts from request schema ("http://" or "https://") in their values. You should specify host without it. Execute command with --verbose for more information or use -h for command for details about arguments'
+    );
```
Comment (Matthiee) [inline src/connector-cli/src/commands/publish/steps/validate-allowed-domains.ts:27]:
```suggestion
    throw new ExecutionError(
      'Some "allowedDomains" start with a request schema ("http://" or "https://"), which is not allowed. You should specify the host without it. Execute the command with --verbose for more information or use -h for command for details about arguments.'
    );
```

File: src/connector-cli/src/commands/publish/steps/update-connector.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  };
+
+  success(`Connector "${result.name}" is updated`, result);
+}
+
+async function getConnectorForUpdate(
+  connectorEndpointBaseUrl: string,
+  token: string
+): Promise<string> {
+  info(`Requesting list of available connectors...`);
```
Comment (Matthiee) [inline src/connector-cli/src/commands/publish/steps/update-connector.ts:83]:
```suggestion
  info(`Requesting the list of available connectors...`);
```

File: src/connector-cli/src/commands/publish/steps/update-connector.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  const connectorsRes = await fetch(connectorEndpointBaseUrl, {
```
Comment (Matthiee) [inline src/connector-cli/src/commands/publish/steps/update-connector.ts:84]: Check if response was OK

File: src/connector-cli/src/commands/publish/steps/update-connector.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    headers: {
+      Authorization: token,
+    },
+  });
+
+  info(
+    `Here are the list of available connectors. Select the one you want to update`
```
Comment (Matthiee) [inline src/connector-cli/src/commands/publish/steps/update-connector.ts:91]:
```suggestion
    `Received the list of available connectors. Please select the one you want to update:`
```

File: src/connector-cli/src/commands/publish/steps/update-connector.ts [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  );
+  const { data: connectors } = await connectorsRes.json();
```
Comment (Matthiee) [inline src/connector-cli/src/commands/publish/steps/update-connector.ts:93]: This is probably not possible, but what if the array of connectors is empty, you'll be stuck in that `while (true)` loop

File: src/connector-cli/src/commands/publish/index.ts [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+  info('Validating runtime options...');
+
+  validateRuntimeOptions(runtimeOptions, config.options);
+
+  info('Extracting package information...');
+
+  const { description, version, apiVersion } =
+    extractPackageInfo(connectorFile);
+
+  info('Compile connector...');
```
Comment (Matthiee) [inline src/connector-cli/src/commands/publish/index.ts:71]:
```suggestion
  info('Compiling connector...');
```

File: src/connector-cli/src/commands/publish/index.ts [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+  const { connectorJs } = await compileConnector(connectorFile);
+
+  // Retrieve capabilities and configurationOptions of the connector
+  // const connectorInfo = await getInfoInternal(compilation);
```
Comment (Matthiee) [inline src/connector-cli/src/commands/publish/index.ts:76]: Commented code

**Review by Matthiee — APPROVED** [review]
LGTM

File: src/connector-cli/src/commands/publish/index.ts (hunk 8 above)
Comment (psamusev) [reply src/connector-cli/src/commands/publish/index.ts:76]: It's here for historical reason, leaved so far

File: src/connector-cli/src/commands/publish/steps/update-connector.ts (hunk 6 above)
Comment (psamusev) [reply src/connector-cli/src/commands/publish/steps/update-connector.ts:93]: You're right, it's not realistic scenario

File: src/connector-cli/src/commands/publish/steps/update-connector.ts (hunk 4 above)
Comment (psamusev) [reply src/connector-cli/src/commands/publish/steps/update-connector.ts]: Done

**Review by Matthiee — APPROVED** [review]

**Pull request merged**